### PR TITLE
Lifetime Model: Deduplicate input file : Closes #6146

### LINKED ISSUE #6146: Misleading warnings when submitting a Lifetime Model exception with an input file containing duplicate entries
### Description

If the user is submitting a Lifetime Model exception request using the CLI and the input file contains some dataset multiple times, then they will see some warning about datasets not existing.

Instead, it should show a warning that the entry is being removed for being a duplicate (so that, when reviewing the summary, one can easily understand the difference between total and successfully-submitted DIDs).

### Steps to reproduce

```console
$ cat covfefe
mc16_13TeV.364677.Sherpa_CT10_CT14nnlo_CSShower_2to2jets_JZ1W.deriv.DAOD_STDM11.e6997_e5984_s3126_r10201_r10210_p4615_tid26084487_00
mc16_13TeV.364677.Sherpa_CT10_CT14nnlo_CSShower_2to2jets_JZ1W.deriv.DAOD_STDM11.e6997_e5984_s3126_r10201_r10210_p4615_tid26084487_00
$ rucio add-lifetime-exception --inputfile covfefe --reason 'For testing purposed; please ignore.' --expiration 2024-01-01
2023-03-23 08:09:13,033	INFO	mc16_13TeV:mc16_13TeV.364677.Sherpa_CT10_CT14nnlo_CSShower_2to2jets_JZ1W.deriv.DAOD_STDM11.e6997_e5984_s3126_r10201_r10210_p4615_tid26084487_00 will be declared
2023-03-23 08:09:13,033	WARNING	mc16_13TeV:mc16_13TeV.364677.Sherpa_CT10_CT14nnlo_CSShower_2to2jets_JZ1W.deriv.DAOD_STDM11.e6997_e5984_s3126_r10201_r10210_p4615_tid26084487_00 does not exist
2023-03-23 08:09:13,066	INFO	Exception successfully submitted. Summary below
Total DIDs                                                                                                2
DID not submitted because it is a file                                                                    0
DID not submitted because it is a container                                                               0
DID not submitted because it is not part of the lifetime campaign                                         0
DID successfully submitted                                                                                1
```

### Rucio Version

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by cserf**

### CODE REVIEW

File: lib/rucio/tests/test_bin_rucio.py [hunk 1]
```diff
@@ -2327,3 +2331,22 @@ def test_admin_rse_update_unsupported_option(self):
...
         assert not err
+
+    @pytest.mark.noparallel(reason='Modify config')
+    def test_lifetime_cli(self):
+        """ CLIENT(USER): Check CLI to declare lifetime exceptions """
+        # Setup data for CLI check
+        tmp_dsn_name = 'container' + rse_name_generator()
+        tmp_dsn_did = self.user + ':' + tmp_dsn_name
+        self.did_client.add_did(scope=self.user, name=tmp_dsn_name, did_type='DATASET')
+        self.did_client.set_metadata(scope=self.user, name=tmp_dsn_name, key='eol_at', value=(datetime.utcnow() - timedelta(seconds=86400)).strftime('%Y-%m-%d'))
```
Comment (dchristidis) [inline lib/rucio/tests/test_bin_rucio.py]: Better to use `days=1` like below.

File: lib/rucio/tests/test_bin_rucio.py [hunk 2]
```diff
@@ -2327,3 +2331,22 @@ def test_admin_rse_update_unsupported_option(self):
...
+        tomorrow = datetime.utcnow() + timedelta(days=1)
+        tomorrow = tomorrow.strftime('%Y-%m-%d')
+        self.config_client.set_config_option(section='lifetime_model', option='cutoff_date', value=tomorrow)
```
Comment (dchristidis) [inline lib/rucio/tests/test_bin_rucio.py]: The first two lines should be merged. The main reason is that, especially now that we’re aiming for type hints, we should avoid changing the type of a variable wherever possible.

But then, I’m wondering whether it is actually necessary to assign this to a variable at all (you don’t do so above).

File: lib/rucio/tests/test_bin_rucio.py [hunk 3]
```diff
@@ -2327,3 +2331,22 @@ def test_admin_rse_update_unsupported_option(self):
...
+        string = codecs.encode("%s\n" % tmp_dsn_did, 'utf-8')
+        with tempfile.NamedTemporaryFile() as fp:
+            fp.write(string)
+            fp.write(string)
+            fp.seek(0)
+            exitcode, out, err = execute("rucio add-lifetime-exception --inputfile %s --reason 'For testing purposed; please ignore.' --expiration 2124-01-01" % fp.name)
```
Comment (dchristidis) [inline lib/rucio/tests/test_bin_rucio.py]: Typo in ‘purposed’ (yes, I am aware that this was *my* typo :smile:).

File: lib/rucio/tests/test_bin_rucio.py [hunk 4]
```diff
@@ -2327,3 +2331,22 @@ def test_admin_rse_update_unsupported_option(self):
...
+        """ CLIENT(USER): Check CLI to declare lifetime exceptions """
+        # Setup data for CLI check
+        tmp_dsn_name = 'container' + rse_name_generator()
+        tmp_dsn_did = self.user + ':' + tmp_dsn_name
+        self.did_client.add_did(scope=self.user, name=tmp_dsn_name, did_type='DATASET')
+        self.did_client.set_metadata(scope=self.user, name=tmp_dsn_name, key='eol_at', value=(datetime.utcnow() - timedelta(seconds=86400)).strftime('%Y-%m-%d'))
+        tomorrow = datetime.utcnow() + timedelta(days=1)
+        tomorrow = tomorrow.strftime('%Y-%m-%d')
+        self.config_client.set_config_option(section='lifetime_model', option='cutoff_date', value=tomorrow)
+        string = codecs.encode("%s\n" % tmp_dsn_did, 'utf-8')
```
Comment (dchristidis) [inline lib/rucio/tests/test_bin_rucio.py]: Could you explain the necessity for using the `codecs` module?

**Review by dchristidis — CHANGES_REQUESTED** [review]
My apologies for being pedantic, but please adhere to the Contributing Guide on how to name commits. May I suggest something like `Lifetime Model: Deduplicate input file; fix #6146`?

In addition, the newly-introduced test case appears to fail quite often.

File: lib/rucio/tests/test_bin_rucio.py (hunk 4 above)
Comment (cserf) [reply lib/rucio/tests/test_bin_rucio.py]: Removed

File: lib/rucio/tests/test_bin_rucio.py [hunk 5]
```diff
@@ -2327,3 +2330,20 @@ def test_admin_rse_update_unsupported_option(self):
...
+    @pytest.mark.noparallel(reason='Modify config')
+    def test_lifetime_cli(self):
+        """ CLIENT(USER): Check CLI to declare lifetime exceptions """
+        # Setup data for CLI check
+        tmp_dsn_name = 'container' + rse_name_generator()
+        tmp_dsn_did = self.user + ':' + tmp_dsn_name
+        self.did_client.add_did(scope=self.user, name=tmp_dsn_name, did_type='DATASET')
+        self.did_client.set_metadata(scope=self.user, name=tmp_dsn_name, key='eol_at', value=(datetime.utcnow() - timedelta(days=1)).strftime('%Y-%m-%d'))
+        self.config_client.set_config_option(section='lifetime_model', option='cutoff_date', value=(datetime.utcnow() + timedelta(days=1)).strftime('%Y-%m-%d'))
+        string = f"%s{tmp_dsn_did}\n"
```
Comment (dchristidis) [inline lib/rucio/tests/test_bin_rucio.py]: There is a leftover `%s` that needs to be removed.

But honestly, I don’t think this variable is helpful. Would you be against `fp.write(f'{tmp_dsn_did}\n' * 2)`?

File: lib/rucio/tests/test_bin_rucio.py [hunk 6]
```diff
@@ -2327,3 +2330,20 @@ def test_admin_rse_update_unsupported_option(self):
...
+        with tempfile.NamedTemporaryFile(mode="w+") as fp:
+            fp.write(string)
+            fp.write(string)
+            fp.seek(0)
+            exitcode, out, err = execute("rucio add-lifetime-exception --inputfile %s --reason 'For testing purpose; please ignore.' --expiration 2124-01-01" % fp.name)
+            assert err.find('does not exist') == -1
```
Comment (dchristidis) [inline lib/rucio/tests/test_bin_rucio.py]: That’s not very Pythonic. How about `assert 'does not exist' not in err`?

**Review by dchristidis — CHANGES_REQUESTED** [review]

File: lib/rucio/tests/test_bin_rucio.py (hunk 5 above)
Comment (cserf) [reply lib/rucio/tests/test_bin_rucio.py]: Fixed

File: lib/rucio/tests/test_bin_rucio.py (hunk 6 above)
Comment (cserf) [reply lib/rucio/tests/test_bin_rucio.py]: Done

**Review by dchristidis — APPROVED** [review]

**Pull request merged**